trappy/cpu_power.py

```diff
@@ -93,6 +93,29 @@ def num_cpus_in_mask(mask):
 
     return bin(value).count("1")
 
+def sanitize_mask(mask):
+    """Return mask string to usable int value """
+    return int(mask.replace(",", ""), 16)
+
+def cpus_to_mask(cpus):
+    """Returns the integer value of the list -> mask translation"""
+    res = 0
+    for cpu in cpus:
+        res |= (1 << cpu)
+    return res
+
+def mask_to_cpus(mask):
+    """Returns the list of the cpus present in the mask"""
+    res = []
+    # Remove '0b' prefix
+    bits = list(bin(mask))[2::]
+    # Reverse to LSB first order
+    bits.reverse()
+    for idx, bit in enumerate(bits):
+        if bit == '1':
+            res.append(idx)
+    return res
```

Comment:
It's not important, but converting to a string does seem a bit laborious.. what about something like this?

```
return [i for i in range(number_of_bits) if mask & (1 << i)]
```
Not sure how you'd get `number_of_bits`, mabye `int(log(mask, 2)) + 1` or something. 

But yeah this doesn't actually matter I'm just being a nerd.

Reply:
I realized devlib has some tooling for this:
```
def mask_to_list(mask):
    size = len(bin(mask)) - 2  # because of "0b"
    return [size - i - 1 for i in xrange(size)
            if mask & (1 << size - i - 1)]
```
This also uses a string conversion though. I also thought of doing it "C-style" but Python has made my laziness even worse...